pkg/gatherers/conditional/gathering_functions.go

```diff
@@ -0,0 +1,52 @@
+package conditional
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// GatheringFunctions is a type to map gathering function name to its params
+type GatheringFunctions = map[GatheringFunctionName]interface{}
```

Comment:
Wouldn't be good to define some type for the params? I guess we could use type embedding and the generic type could have hopefully something common for all the params... I guess name at least.

Reply:
It seems redundant to include all the possible things in one type

Reply:
Well I've been thinking about some inheritance (Java-ish :)) structure (in Go it's more composition though), but it looks like the params (e.g `GatherLogsOfNamespaceParams` and `AlertIsFiringConditionParams`) have nothing in common. So I guess let's keep it as it is.....but it's interesting problem... :)